editor/test/integration/editor_extensions_test.clj

```diff
@@ -0,0 +1,289 @@
+                                                         (throw (LuaError. "failed immediately")))
+                            "suspend_fail_async" (rt/suspendable-lua-fn [_]
+                                                   (future/failed (LuaError. "failed async")))})
+          lua-fn (->> (rt/read "return function()
+                                  local success1, value1 = pcall(suspend_fail_immediately)
+                                  local success2, value2 = pcall(suspend_fail_async)
+                                  return {
+                                    {success1, value1},
+                                    {success2, value2},
+                                  }
+                                end")
+                      (rt/bind rt)
+                      (rt/invoke-immediate rt))]
+      (is (= [[false "failed immediately"]
+              [false "failed async"]]
+             (rt/->clj rt @(rt/invoke-suspending rt lua-fn)))))))
+
+(deftest immediate-failures-test
+  (test-util/with-loaded-project
+    (let [rt (rt/make :env {"immediate_error" (rt/lua-fn [_]
+                                                (throw (Exception. "fail")))})]
+      (is
+        (= [false "fail"]
+           (->> (rt/read "local success1, result1 = pcall(immediate_error)
+                          return {success1, result1}")
+                (rt/bind rt)
+                (rt/invoke-immediate rt)
+                (rt/->clj rt)))))))
+
+(deftype StaticSelection [selection]
+  handler/SelectionProvider
+  (selection [_] selection)
+  (succeeding-selection [_] [])
+  (alt-selection [_] []))
+
+(deftest editor-scripts-commands-test
+  (test-util/with-loaded-project "test/resources/editor_extensions/commands_project"
+    (let [resource-sync (bound-fn* workspace/resource-sync!)
+          sprite-outline (:node-id (test-util/outline (test-util/resource-node project "/main/main.collection") [0 0]))]
+      (extensions/reload! project :all
+                          :reload-resources (fn []
+                                              (resource-sync workspace)
+                                              (future/completed nil))
+                          :can-execute? (constantly (future/completed true))
+                          :display-output println)
+      (let [handler+context (handler/active
+                              (:command (first (handler/realize-menu :editor.outline-view/context-menu-end)))
+                              (handler/eval-contexts
+                                [(handler/->context :outline {} (->StaticSelection [sprite-outline]))]
+                                false)
+                              {})]
+        (is (= [0.0 0.0 0.0] (test-util/prop sprite-outline :position)))
+        (is (= 1.0 (test-util/prop sprite-outline :playback-rate)))
+        (is (some? handler+context))
+        (is (handler/enabled? handler+context))
+        (is (any? @(handler/run handler+context)))
```

Comment:
Does `any?` do anything? Should we perhaps check that it returns an expected value instead? Even `some?` or `nil?` would be preferable to `any?`, right? 😅

Reply:
I typically do `(is (= expected-value ...))`, or `(is (nil? ...))` if it isn't supposed to return anything.

I think `(is (some? ...))` is preferable to `(is (any? ...))`, because at least it will tell you if the call is suddenly returning something unexpected.

But I despise the mere existence of `any?`, so I want to pretend it doesn't exist. 😅

Reply:
I like `any?` when it's used as a default predicate that always succeeds. It's like `(constantly true)`, but without creating new variadic functions. I wish there was also `none?`... Anyway, I'll fix the test so it's clear that we are testing that it does not throw.